test_common/harness/kernelHelpers.cpp

```diff
@@ -809,6 +811,56 @@ int create_single_kernel_helper(cl_context context,
                                 const char *buildOptions,
                                 const bool openclCXX)
 {
+
+    // Check the build options for the -cl-std option.
+    if (!buildOptions || !strstr(buildOptions, "-cl-std"))
+    {
+        // If the build option isn't present add it using the latetest OpenCL-C
+        // version supported by the device. This allows calling code to force a
+        // particular CL C version if it is required, but also means that
+        // callers need not specify a version if they want to assume the most
+        // recent CL C.
+
+        // First get all the devices in the context and find the minumum
+        // supported OpenCL C version.
+        size_t devices_size_in_bytes{};
+        int error = clGetContextInfo(context, CL_CONTEXT_DEVICES, 0, nullptr,
+                                     &devices_size_in_bytes);
+        test_error(error, "clGetDeviceInfo failed for CL_CONTEXT_DEVICES");
+        std::vector<cl_device_id> devices(devices_size_in_bytes
+                                          / sizeof(cl_device_id));
+        error =
+            clGetContextInfo(context, CL_CONTEXT_DEVICES, devices_size_in_bytes,
+                             devices.data(), nullptr);
+        Version min_version{ (std::numeric_limits<int>::max)(),
+                             (std::numeric_limits<int>::max)() };
+        std::for_each(devices.begin(), devices.end(),
+                      [&min_version](cl_device_id device) {
+                          auto version = get_device_cl_version(device);
+                          min_version = (std::min)(version, min_version);
```

Comment:
Is the logic not reversed here?

If the intent is to select the latest/newest CL C version, then this loop should effectively be `max_version` instead?

Reply:
Resolving, the lowest common denominator is being chosen.

This raises a different question though which I'll post a new comment for.